File: Practice/AllDomains/CoreCS/DataStructures/Arrays/SparseArrays.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int n = Convert.ToInt32(Console.ReadLine().Trim());
        string [] strings = new string[n];
        for(int i = 0; i < n; i++)
        {
            strings[i] = Console.ReadLine().Trim();
        }
        int q = Convert.ToInt32(Console.ReadLine().Trim());
        string [] queries = new string[q];
        for(int i = 0; i < q; i++)
        {
            queries[i] = Console.ReadLine().Trim();
        }

        for(int j = 0; j < q; j++)
        {
            int counter = 0;
            string query = queries[j];
            for(int i = 0; i < n; i++)
            {
                if(strings[i] == query) counter++;
            }
            Console.WriteLine(counter);
        }
    }
}

File: Practice/AllDomains/CoreCS/DataStructures/Arrays/SparseArrays.js
/*	author:@shivkrthakur  */
var index = 0;

function readLine(input){
    return input[index++];
}

function processData(input) {
    //Enter your code here
    var n = parseInt(readLine(input));
    var inputArr = [];
    for(var i = 0; i < n; i++){
        inputArr[i] = readLine(input);
    }
    var q = parseInt(readLine(input));
    for(var j = 0; j < q; j++){
        var src = readLine(input);
        var count = 0;
        for(var p = 0; p < n; p++){
            if(inputArr[p] === src) count++;
        }
        console.log(count);
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input.split('\n'));
});
